### PR TITLE
Documentation for DocumentReference Search

### PR DESCRIPTION
**Pull request by averyallen**
Updates the DocumentReference resource for upcoming search functionality.

Also adds a note to one of the DocumentReference create fields describing an additional requirement around meta tags.

### CODE REVIEW

File: content/millennium/dstu2/infrastructure/document-reference.md [hunk 1]
```diff
@@ -9,32 +9,95 @@ title: DocumentReference | DSTU 2 API
...
+
+### Parameters
+
+ Name           | Required?          | Type          | Description
+----------------|--------------------|---------------|-----------------------------------------------------------------------------------------------
+`patient`       | This, or `subject` | [`reference`] | The identifier of a patient to list references for. Example: `patient=12345`
+`subject`       | This, or `patient` | [`reference`] | The subject of the document reference. Must represent a Patient resource. May use the :Patient modifier. Example: `subject=Patient/1316020 or subject:Patient=1316020`
+`encounter`     | No                 | [`reference`] | The encounter which belongs to the reference.  Must represent an Encounter resource.  May include a single or comma separated list of references. Example: `encounter=1353`
+`created`       | No                 | [`date`]      | A date/time the referenced document was created.  Must use the `ge` and `le` prefixes.  Example: `created=le2017-01-5&created=ge2017-02-7`
+[`_count`]      | No                 | [`number`]    | The maximum number of results to return. Defaults to `10`.
```
Comment (MaxPhilips) [inline content/millennium/dstu2/infrastructure/document-reference.md]: Don't add default value

File: content/millennium/dstu2/infrastructure/document-reference.md [hunk 2]
```diff
@@ -9,32 +9,95 @@ title: DocumentReference | DSTU 2 API
...
+
+<%= authorization_types(practitioner: true, patient: true, system: true) %>
+
+### Parameters
+
+ Name           | Required?          | Type          | Description
+----------------|--------------------|---------------|-----------------------------------------------------------------------------------------------
+`patient`       | This, or `subject` | [`reference`] | The identifier of a patient to list references for. Example: `patient=12345`
+`subject`       | This, or `patient` | [`reference`] | The subject of the document reference. Must represent a Patient resource. May use the :Patient modifier. Example: `subject=Patient/1316020 or subject:Patient=1316020`
+`encounter`     | No                 | [`reference`] | The encounter which belongs to the reference.  Must represent an Encounter resource.  May include a single or comma separated list of references. Example: `encounter=1353`
```
Comment (MaxPhilips) [inline content/millennium/dstu2/infrastructure/document-reference.md]: "The encounter which belongs to the reference." sounds a little off, is that supposed to be "The encounter which belongs to the **document** reference."?

Or even "The encounter to which the **document** reference belongs."?

File: content/millennium/dstu2/infrastructure/document-reference.md [hunk 3]
```diff
@@ -9,32 +9,95 @@ title: DocumentReference | DSTU 2 API
...
+
+### Authorization Types
+
+<%= authorization_types(practitioner: true, patient: true, system: true) %>
+
+### Parameters
+
+ Name           | Required?          | Type          | Description
+----------------|--------------------|---------------|-----------------------------------------------------------------------------------------------
+`patient`       | This, or `subject` | [`reference`] | The identifier of a patient to list references for. Example: `patient=12345`
```
[the remark this answers is not in the document]
Comment (averyallen) [reply content/millennium/dstu2/infrastructure/document-reference.md]: How do you feel about "The patient the document reference is about?"

[the remark this answers is not in the document]
Comment (MadhurRaj) [reply content/millennium/dstu2/infrastructure/document-reference.md]: I'm good with that

**Comment by MaxPhilips** [comment]
+1

**Pull request merged**